src/utils/UrlUtils.ts

```diff
@@ -55,10 +55,26 @@ export class UrlUtils {
     return {
       pathsNormalized,
       queryNormalized,
-      path: this.addToUrlIfNotEmpty(pathsNormalized, '/', '')
+      path: this.addToUrlIfNotEmpty(pathsNormalized, '/', UrlUtils.getRelativePathLeadingCharacters(toNormalize))
     };
   }
 
+  private static getRelativePathLeadingCharacters(toNormalize: IUrlNormalize) {
+    let leadingRelativeUrlCharacters = '';
+
+    const relativeUrlLeadingCharactersRegex = /^(([\/])+)/;
```

Comment:
Are the parentheses/square brackets needed or could the regex be `/^\/+/` if the goal is to detect the leading slashes?